### PR TITLE
Delete the SkPath/SkRRect code duplication in for physical model layers.

### PR DESCRIPTION
**Pull request by amirh**
Just always use SkPath.

This is a breaking change, framework PR: https://github.com/flutter/flutter/pull/13942.
Will announce a breaking change before submitting.

### CODE REVIEW

**Comment by Hixie** [comment]
When landing this on the framework please keep an eye on the performance numbers to make sure this doesn't cause any regressions. Thanks.

File: flow/layers/physical_model_layer.h [hunk 1]
```diff
@@ -39,88 +50,11 @@ class PhysicalModelLayer : public ContainerLayer {
...
-
-#if defined(OS_FUCHSIA)
-  // Scenic does not currently support compositing arbitrary shaped layers,
-  // so we just use the path's bounding rectangle.
-  // |flow::PhysicalLayerShape|
-  const SkRRect& getFrameRRect() const override { return frameRRect_; }
-#endif  // defined(OS_FUCHSIA)
- private:
   SkPath path_;
+  bool isRect_;
```
Comment (jason-simmons) [inline flow/layers/physical_model_layer.h]: add a constructor initializer for isRect_

File: flow/layers/physical_model_layer.h [hunk 2]
```diff
@@ -16,9 +16,20 @@ class PhysicalModelLayer : public ContainerLayer {
...
 
-  void set_shape(std::unique_ptr<PhysicalLayerShape> shape) {
-    shape_ = std::move(shape);
+  void set_path(const SkPath& path) {
+    path_ = path;
+#if defined(OS_FUCHSIA)
+    SkRRect rrect = SkRRect::MakeEmpty();
+    if (path.isRRect(&rrect)) {
+      frameRRect_ = rrect;
+      isRect_ = rrect.isRect();
```
Comment (jason-simmons) [inline flow/layers/physical_model_layer.h]: I think we'll want the isRect check on all platforms so Paint() can avoid the canvas.saveLayer() call in the common case of a rectangular shape

Comment (amirh) [reply flow/layers/physical_model_layer.h]: good catch

File: flow/layers/physical_model_layer.h (hunk 1 above)
Comment (amirh) [reply flow/layers/physical_model_layer.h]: done

File: flow/layers/physical_model_layer.h [hunk 3]
```diff
@@ -13,19 +13,22 @@ class PhysicalLayerShape;
...
     if (path.isRRect(&rrect)) {
-      frameRRect_ = rrect;
       isRect_ = rrect.isRect();
     } else {
       isRect_ = false;
-      frameRRect_ = SkRRect::MakeRect(path.getBounds());
+    }
+#if defined(OS_FUCHSIA)
+    if (path.isRRect(&rrect)) {
+      frameRRect_ = rrect;
```
Comment (jason-simmons) [inline flow/layers/physical_model_layer.h]: Now it looks like isRect_ will not be set to true in the Fuchsia case.  I'd recommend moving this to physical_model_layer.cc and having one code path for all platforms that calls path.isRRect(&frameRRect_) and sets isRect_ accordingly

**Comment by amirh** [comment]
PTAL, also renamed physical model to physical shape.

File: flow/layers/physical_model_layer.cc [hunk 4]
```diff
@@ -9,6 +9,11 @@
 
 namespace flow {
 
+PhysicalModelLayer::PhysicalModelLayer() {
+  isRect_ = false;
+  frameRRect_ = SkRRect::MakeEmpty();
```
Comment (jason-simmons) [inline flow/layers/physical_model_layer.cc]: No need to do this explicitly - SkRRect's default constructor will make it empty

**Pull request merged**